### PR TITLE
Use find_by instead of a dynamic finder

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

File: hydra-access-controls/lib/hydra/user.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@ module ClassMethods
     # By default, uses the unique identifier specified in by devise authentication_keys (ie. find_by_id, or find_by_email).  
     # You must have that find method implemented on your user class, or must override find_by_user_key
```
Comment (atz) [inline hydra-access-controls/lib/hydra/user.rb]: Need to update these comments that are no longer instructional.  They also may reveal the motivation of using the more particular method name (so as to allow overriding but not to mess with the core `find_by`).

**Comment by jeremyf** [comment]
Ping @atx

**Comment by jeremyf** [comment]
@jcoyne There is now a merge conflict

And repinging @atz

**Comment by mjgiarlo** [comment]
@jcoyne @jeremyf I resolved the merge conflict.

**Comment by jcoyne** [comment]
@mjgiarlo I redid the rebase (the build was failing)

**Comment by mjgiarlo** [comment]
@atz It looks like @jcoyne has addressed your review. Can you re-review?

**Pull request merged**